presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -840,12 +844,15 @@ private void invalidatePartitionCache(String databaseName, String tableName)
                         }
                     }));
         }
-        catch (NoSuchObjectException e) {
+        catch (HiveMetastoreClientNoSuchObjectException e) {
             // assume none of the partitions in the batch are available
             return stream(partitionNames.spliterator(), false)
                     .collect(toMap(identity(), (name) -> Optional.empty()));
         }
-        catch (TException e) {
+        catch (HiveMetastoreClientException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (IOException e) {
```

Comment:
What throws `IOException`? Same comment applies to the blocks below that add `IOException`